src/videojs-hls.js

```diff
@@ -881,7 +887,19 @@ videojs.Hls.getPlaylistDuration = function(playlist, startIndex, endIndex) {
 
   for (; i >= startIndex; i--) {
     segment = playlist.segments[i];
-    dur += (segment.duration !== undefined ? segment.duration : playlist.targetDuration) || 0;
+    if (segment.preciseTimestamp && segment.preciseTimestamp >= 0) {
+      var offset = 0;
```

Comment:
could we calculate the `preciseDuration` statically instead of having to re-calculate it every time?
Or perhaps, we could store it into a `preciseDuration` after the first calculation.